### PR TITLE
Feature/EUI-3936 choose exclusion option

### PR DESCRIPTION
**Pull request by rayliangatos**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-3936
https://tools.hmcts.net/jira/browse/EUI-3944

### Change description ###
Choose exlusion and chosse role

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by leepavlouhmcts — APPROVED** [review]

File: src/role-access/containers/exclusion/choose-person-role/choose-person-role.component.ts [hunk 1]
```diff
@@ -1,46 +1,91 @@
...
+    this.radioOptionControl = new FormControl(this.personRole ? this.personRole : '', [Validators.required]);
+    this.formGroup = new FormGroup({[this.radioControlName]: this.radioOptionControl});
+
     this.roles$ = this.roleExclusionsService.getRolesCategory();
     this.roles$.subscribe((roles) => {
-      this.roleOptions = [];
-      roles.forEach(role => {
-        const currentOption: RadioOption = { radioId: role.roleId, radioName: role.roleName };
-        this.roleOptions.push(currentOption);
+      this.optionsList = roles.map(role => {
```
Comment (kuda1992) [inline src/role-access/containers/exclusion/choose-person-role/choose-person-role.component.ts:54]: Could be inline..

File: src/role-access/containers/exclusion/exclusion-home/exclusion-home.component.ts [hunk 2]
```diff
@@ -131,14 +139,19 @@ export class ExclusionHomeComponent implements OnInit, OnDestroy {
         }
         break;
       }
+      case ExclusionNavigationEvent.CANCEL:
+        this.router.navigateByUrl(`cases/case-details/${this.caseId}`).then(r => {
```
Comment (kuda1992) [inline src/role-access/containers/exclusion/exclusion-home/exclusion-home.component.ts:143]: No need to call the promise method here

**Review by kuda1992 — CHANGES_REQUESTED** [review]
Looks good just one minor change

File: src/role-access/containers/exclusion/exclusion-home/exclusion-home.component.ts (hunk 2 above)
Comment (rayliangatos) [reply src/role-access/containers/exclusion/exclusion-home/exclusion-home.component.ts:143]: This is to suppress the ts warning

File: src/role-access/containers/exclusion/choose-person-role/choose-person-role.component.ts (hunk 1 above)
Comment (rayliangatos) [reply src/role-access/containers/exclusion/choose-person-role/choose-person-role.component.ts:54]: this has been indented correctly, it aligns with the original roles.forEach...

**Review by kuda1992 — APPROVED** [review]

**Pull request merged**